### DISCUSSION TITLE
Why are unsigned ints being used here?

### DISCUSSION THREAD

**Discussion by LucHarding** [hathach/tinyusb#1922] [Q&A]
So I'm working with USBTMC, and I am currently having problems running some tests against my code. My code is using this structure from line 73 - 81 of the `usbtmc.h` header [https://github.com/hathach/tinyusb/blob/master/src/class/usbtmc/usbtmc.h#L73](url)

My question is, why is this line:
`unsigned int EOM  : 1         ; ///< EOM set on last byte`
using an unsigned int, and not a uint8_t? Because when I run my tests with a 64bit compiler, the static assertions fail? Is this just an oversight, or is there something else that is happening here that I don't understand.

**Comment by cr1901** [comment]
According to [blame](https://github.com/hathach/tinyusb/commit/c755aee7d0c1e2486055e02ac353878d8da3ba59) it _was_ at once point a `uint8_t`. Idk if @pigrew is around, but perhaps he remembers why he changed it?
